modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -357,4 +368,46 @@ Map<String, Long> toMap() {
             return results;
         }
     }
+
+    private static final TimeValue RETRY_STATS_WINDOW = TimeValue.timeValueMinutes(5);
+
+    private static class AwsApiRetryStats {
+        private final LongAdder requests = new LongAdder();
+        private final LongAdder retries = new LongAdder();
+        private final LongAdder throttles = new LongAdder();
+    }
+
+    private volatile AwsApiRetryStats awsApiRetryStats = new AwsApiRetryStats();
+
+    private void maybeLogRetryStats() {
+        var stats = awsApiRetryStats;
+        awsApiRetryStats = new AwsApiRetryStats();
+        long requests = stats.requests.sum();
+        long retries = stats.retries.sum();
+        long throttles = stats.throttles.sum();
+        boolean retried = retries > 0 || throttles > 0;
+        if (retried || logger.isDebugEnabled()) {
+            var message = new ESLogMessage(
+                "Recorded {} requests including {} retries and {} throttles within {}",
```

Comment:
I think we want to know the distribution of retry-count-until-success here. Did every request either succeed on the first attempt except for a few that took 10 goes? Or did we retry them all exactly once?